src/Archives/ClmFile.cpp

```diff
@@ -54,9 +54,9 @@ namespace Archives
 	// Extracts the internal file corresponding to index
 	void ClmFile::ExtractFile(std::size_t index, const std::string& pathOut)
 	{
-		VerifyIndexInBounds(index);
+		const auto& indexEntry = indexEntries.at(index);
```

Comment:
While I'm not a fan of removing the custom error messaging, we could still return a reference here to reduce index lookups.